astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,380 @@
+    """
+    Convert the table metadata for time informational keywords
+    to astropy Time.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time columns are to be converted.
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Read in Global Informational keywords as Time
+    for key, value in global_info.items():
+        # FITS uses a subset of ISO-8601 for DATE-xxx
+        if key.startswith('date'):
+            if key not in table.meta:
+                precision = (lambda x: len(x.split('.')[1])
+                             if '.' in x else 0)
+                scale = 'utc' if key == 'date' else global_info['scale']
+                if scale is not None:
+                    try:
+                        table.meta[key.upper()] = Time(value, scale=scale,
+                                                   precision=precision(value))
+                        continue
+                    except ValueError:
+                        pass
+                table.meta[key.upper()] = value
+
+        # MJD-xxx in MJD according to TIMESYS
+        elif key.startswith('mjd-'):
+            if key not in table.meta:
+                if global_info['scale'] is not None:
+                    try:
+                        table.meta[key.upper()] = Time(value, scale=global_info['scale'],
+                                                       format='mjd')
+                        continue
+                    except ValueError:
+                        pass
+                table.meta[key.upper()] = value
+
+
+def _convert_time_columns(table, global_info, time_columns):
+    """
+    Convert time columns to astropy Time columns.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time columns are to be converted.
+    global_info : dict
+        Global time reference frame information.
+    time_columns : dict
+        Column-specific time reference frame override information.
+    """
+
+    # The code might fail while attempting to read FITS files not written by astropy.
+
+    # Read in time coordinate columns as Time
+    for idx, time_col in time_columns.items():
+        time_colname = table.colnames[idx - 1]
+        if table[time_colname].shape[-1] == 2 and table[time_colname].ndim > 1:
+            table[time_colname] = Time(table[time_colname][..., 0],
+                                       table[time_colname][..., 1],
+                                       format='jd', scale=time_col['scale'])
+        else:
+            warnings.warn(
+                'Time column {} is not in the astropy required (jd1, jd2) format. '
+                'Hence, it will not be read as astropy Time'.format(col.info.name),
+                AstropyUserWarning)
+            continue
+        # Add location if attribute ``pos`` is set
+        try:
+            if time_col['pos'] == 'TOPOCENTER':
```

Comment:
I think it would make more sense to do this in `_verify_column`. Then you do not need `global_info` in this function any more.